### PR TITLE
VideoProcessors v3.0

### PR DESCRIPTION
**Pull request by manjeshbhargav**
**Contributing to Twilio**

> All third party contributors acknowledge that any contributions they provide will be made under the same open source license that the open source project is provided under.

- [ ] I acknowledge that all my contributions will be made under the project's license.

### Description

A description of what this PR does.

### Before review
* [x] Updated CHANGELOG.md if necessary
* [x] Added unit tests if necessary
* [x] Updated affected documentation
* [ ] Verified locally with `npm test`
* [ ] Manually sanity tested running locally
* [ ] Included screenshot as PR comment (if needed)
* [x] Ready for review

### CODE REVIEW

File: tsdef/VideoProcessor.d.ts [hunk 1]
```diff
@@ -1,6 +1,33 @@
+/**
+ * @beta VideoFrame is part of Video Processor V3 support which is currently in beta.
```
Comment (manjeshbhargav) [inline tsdef/VideoProcessor.d.ts:2]: Remove beta tag

File: tsdef/VideoProcessor.d.ts [hunk 2]
```diff
@@ -1,6 +1,33 @@
...
+  displayHeight: number;
+  displayWidth: number;
+  timestamp: number;
+  constructor(videoSource: HTMLVideoElement);
+  close(): void;
+}
+
+/**
+ * A processor for manipulating video frames.
+ * @beta Video Processor V3 support is currently in beta.
```
Comment (manjeshbhargav) [inline tsdef/VideoProcessor.d.ts:14]: Remove beta tag

File: tsdef/VideoProcessor.d.ts (hunk 1 above)
Comment (manjeshbhargav) [reply tsdef/VideoProcessor.d.ts:2]: These APIs are used by other developers apart from Video Processors SDK, so let's remove references to the video processors sdk.

File: tsdef/VideoProcessor.d.ts (hunk 2 above)
Comment (manjeshbhargav) [reply tsdef/VideoProcessor.d.ts:14]: These APIs are used by other developers apart from Video Processors SDK, so let's remove references to the video processors sdk.

File: tsdef/VideoProcessor.d.ts [hunk 3]
```diff
@@ -1,6 +1,33 @@
...
+  /**
+   * Process a video frame.
+   * @param inputFrameBuffer The input frame buffer. Can be one of:
+   * - OffscreenCanvas
+   * - HTMLCanvasElement
+   * - HTMLVideoElement
+   * - VideoFrame (beta)
+   * @param outputFrameBuffer The output frame buffer. The context type can be:
+   * - '2d' (default)
+   * - 'bitmaprenderer' (beta, Chromium-based browsers only)
```
Comment (manjeshbhargav) [inline tsdef/VideoProcessor.d.ts:26]: Remove the text in brackets

File: tsdef/types.d.ts [hunk 4]
```diff
@@ -22,7 +22,16 @@ export type DataTrackPublication = LocalDataTrackPublication | RemoteDataTrackPu
 export type VideoTrackPublication = LocalVideoTrackPublication | RemoteVideoTrackPublication;
 
 export interface AddProcessorOptions {
-  inputFrameBufferType?: 'video' | 'canvas' | 'offscreencanvas';
+  /**
+   * The type of input frame buffer to use.
+   * @beta The 'videoframe' value is currently in beta and only supported on browsers that support VideoFrame API
```
Comment (manjeshbhargav) [inline tsdef/types.d.ts:27]: Remove line

File: tsdef/types.d.ts [hunk 5]
```diff
@@ -22,7 +22,16 @@ export type DataTrackPublication = LocalDataTrackPublication | RemoteDataTrackPu
...
+   */
+  inputFrameBufferType?: 'video' | 'canvas' | 'offscreencanvas' | 'videoframe';
+
+  /**
+   * The type of context to use for the output frame buffer.
+   * @beta The 'bitmaprenderer' value is currently in beta. It is only supported on Chromium-based browsers, and falls back to '2d' on other browsers.
```
Comment (manjeshbhargav) [inline tsdef/types.d.ts:33]: Remove line

File: tsdef/VideoProcessor.d.ts (hunk 3 above)
Comment (luisrivas) [reply tsdef/VideoProcessor.d.ts:26]: You meant in parentheses, right?

File: tsdef/VideoProcessor.d.ts (hunk 1 above)
Comment (luisrivas) [reply tsdef/VideoProcessor.d.ts]: Done 👍

File: tsdef/VideoProcessor.d.ts (hunk 2 above)
Comment (luisrivas) [reply tsdef/VideoProcessor.d.ts]: Done 👍

File: tsdef/types.d.ts (hunk 4 above)
Comment (luisrivas) [reply tsdef/types.d.ts]: Done 👍

File: tsdef/types.d.ts (hunk 5 above)
Comment (luisrivas) [reply tsdef/types.d.ts]: Done 👍

**Review by luisrivas — APPROVED** [review]

**Review by charliesantos — APPROVED** [review]
Looks great!

**Pull request merged**